streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -149,97 +134,94 @@ public String toString(final String indent) {
+            }
+
+            StreamsMetricsImpl.addInvocationRateAndCount(processRateSensor, STREAM_PROCESSOR_NODE_METRICS, tagMap, PROCESS);
+        }
+
+        Sensor processRateSensor() {
+            return processRateSensor;
+        }
+
+        public Sensor suppressionEmitRateSensor() {
+            if (suppressionEmitRateSensor == null) {
+                suppressionEmitRateSensor = metrics.nodeLevelSensor(SUPPRESSION_EMIT_RECORDS, processorNodeName, taskName, Sensor.RecordingLevel.DEBUG);
+                StreamsMetricsImpl.addInvocationRateAndCount(suppressionEmitRateSensor, STREAM_PROCESSOR_NODE_METRICS, tagMap, SUPPRESSION_EMIT_RECORDS);
+            }
+
+            return suppressionEmitRateSensor;
         }
 
-        private void removeAllSensors() {
-            metrics.removeAllNodeLevelSensors(taskName, processorNodeName);
+        public Sensor droppedLateRecordsRateSensor() {
+            if (droppedLateRecordsRateSensor == null) {
+                droppedLateRecordsRateSensor = metrics.nodeLevelSensor(DROPPED_LATE_RECORDS, processorNodeName, taskName, Sensor.RecordingLevel.INFO);
+                StreamsMetricsImpl.addInvocationRateAndCount(droppedLateRecordsRateSensor, STREAM_PROCESSOR_NODE_METRICS, tagMap, DROPPED_LATE_RECORDS);
+            }
+
+            return droppedLateRecordsRateSensor;
         }
 
-        private static Sensor createTaskAndNodeLatencyAndThroughputSensors(final String operation,
-                                                                           final StreamsMetricsImpl metrics,
-                                                                           final String taskName,
-                                                                           final String processorNodeName,
-                                                                           final Map<String, String> taskTags,
-                                                                           final Map<String, String> nodeTags) {
-            final Sensor parent = metrics.taskLevelSensor(taskName, operation, Sensor.RecordingLevel.DEBUG);
-            addAvgMaxLatency(parent, PROCESSOR_NODE_METRICS_GROUP, taskTags, operation);
-            addInvocationRateAndCount(parent, PROCESSOR_NODE_METRICS_GROUP, taskTags, operation);
-
-            final Sensor sensor = metrics.nodeLevelSensor(taskName, processorNodeName, operation, Sensor.RecordingLevel.DEBUG, parent);
-            addAvgMaxLatency(sensor, PROCESSOR_NODE_METRICS_GROUP, nodeTags, operation);
-            addInvocationRateAndCount(sensor, PROCESSOR_NODE_METRICS_GROUP, nodeTags, operation);
-
-            return sensor;
+        public Sensor skippedRecordsRateSensor() {
+            if (skippedRecordsRateSensor == null) {
+                // keep the task-level parent sensor
+                final Sensor taskLevelSensor = metrics.taskLevelSensor(SKIPPED_RECORDS, taskName, Sensor.RecordingLevel.INFO);
```

Comment:
Yes it will be cleared at the task-level `TaskMetrics`.

More specifically, at this place the task sensor should always be created already, and hence to be on the safer side, we should to something like `getButNotCreate` apis.